crates/nix_rs/src/config.rs

```diff
@@ -59,9 +62,21 @@ impl NixConfig {
     pub async fn from_nix(
         nix_cmd: &super::command::NixCmd,
     ) -> Result<NixConfig, super::command::NixCmdError> {
-        let v = nix_cmd
-            .run_with_args_expecting_json(&["show-config", "--json"])
-            .await?;
+        let nix_version = NixVersion::from_nix().await?;
```

Comment:
This doesn't seem like a very good approach to me, still thinking how this can be handled better.

Reply:
It may well require some refactoring outside of the context of this code. I'll do that.